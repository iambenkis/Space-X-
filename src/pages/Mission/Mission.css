.table {
  border-collapse: collapse;
  font-size: 1.4rem;
  width: 95%;
  margin: 3rem auto;
}

.tableData {
  border: 2px solid #ddd;
  padding: 1rem;
}

.tableData:first-child {
  vertical-align: text-top;
}

.tableHeading {
  border: 2px solid #ddd;
  padding: 0.5rem 1rem;
  text-align: left;
}

.missionName {
  color: rgb(63, 63, 63);
  font-weight: 600;
}

.description {
  font-size: 1.25rem;
}

.missionButton {
  text-decoration: none;
  padding: 0.6rem;
  background-color: transparent;
  border: 2px solid rgb(63, 63, 63);
  color: rgb(63, 63, 63);
  font-weight: 600;
  cursor: pointer;
}

.label {
  width: 12rem;
  text-align: center;
}

.missionButton.joined {
  border: 2px solid orangered;
  color: orangered;
}

.notAMember {
  background-color: rgb(63, 63, 63);
  color: white;
  padding: 0.4rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
}

.activeMember {
  background-color: rgb(0, 119, 255);
  color: white;
  font-size: 1rem;
  padding: 0.4rem 0.6rem;
  border-radius: 2rem;
}
