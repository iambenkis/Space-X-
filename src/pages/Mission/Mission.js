import React from 'react';
import { useSelector } from 'react-redux';
// import { nanoid } from '@reduxjs/toolkit';

const Mission = () => {
  const state = useSelector((state) => state.missions);
  return (
    <>
      <table className="table">
        <thead className="tableHeading">
          <tr className="tableRow">
            <th className="tableHeading">Mission</th>
            <th className="tableHeading">Description</th>
            <th className="tableHeading">Status</th>
            <th className="tableHeading">{' '}</th>
          </tr>
        </thead>
        <tbody className="tableBody">
          {
            state.payload.map((mission) => (
              <tr key={mission.mission_id} className="tableRow">
                <td className="tableData">{mission.mission_name}</td>
                <td className="tableData">{mission.description}</td>
                <td className="tableData">
                  {
                    mission.reserved
                      ? 'Active Member'
                      : 'Not A Member'
                  }
                </td>
                <td className="tableData">
                  <button type="button">Join Missio</button>
                </td>
              </tr>
            ))
          }
        </tbody>
      </table>
    </>
  );
};

export default Mission;
