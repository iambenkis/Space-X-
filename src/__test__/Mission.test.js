import React from 'react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/Rockets/configurationStore';
import Mission from '../pages/Mission/Mission';

describe('Test Mission component ', () => {
  test('Does Mission match the snapshot ?', () => {
    const missionTree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Mission />
        </Provider>
      </BrowserRouter>,
    );
    expect(missionTree).toMatchSnapshot();
  });

  test('Does Mission match the snapshot ?', () => {
    const missionTree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Mission />
        </Provider>
      </BrowserRouter>,
    );
    expect(missionTree).toMatchSnapshot();
  });
});
