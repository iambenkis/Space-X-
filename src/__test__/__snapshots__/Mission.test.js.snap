// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Mission component  Does Mission match the snapshot ? 1`] = `
<table
  className="table"
>
  <thead
    className="tableHeading"
  >
    <tr
      className="tableRow"
    >
      <th
        className="tableHeading"
      >
        Mission
      </th>
      <th
        className="tableHeading"
      >
        Description
      </th>
      <th
        className="tableHeading"
      >
        Status
      </th>
      <th
        className="tableHeading"
      />
    </tr>
  </thead>
  <tbody
    className="tableBody"
  />
</table>
`;

exports[`Test Mission component  Does Mission match the snapshot ? 2`] = `
<table
  className="table"
>
  <thead
    className="tableHeading"
  >
    <tr
      className="tableRow"
    >
      <th
        className="tableHeading"
      >
        Mission
      </th>
      <th
        className="tableHeading"
      >
        Description
      </th>
      <th
        className="tableHeading"
      >
        Status
      </th>
      <th
        className="tableHeading"
      />
    </tr>
  </thead>
  <tbody
    className="tableBody"
  />
</table>
`;
